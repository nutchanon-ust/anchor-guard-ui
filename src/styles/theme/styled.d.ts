import 'styled-components';
import { Theme } from './themes';

/* This is the suggested way of declaring theme types */
declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}
