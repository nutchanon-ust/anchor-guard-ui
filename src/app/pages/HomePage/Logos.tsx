import * as React from 'react';
import styled from 'styled-components/macro';
import { ReactComponent as CRALogo } from './assets/cra-logo.svg';
import { ReactComponent as RPLogo } from './assets/rp-logo.svg';
import { ReactComponent as PlusSign } from './assets/plus-sign.svg';

export function Logos() {
  return (
    <Wrapper>
      <CRALogo className="logo" />
      <PlusSign className="sign" />
      <RPLogo className="logo" />
    </Wrapper>
  );
}

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  color: ${p => p.theme.border};

  .logo {
    width: 4.5rem;
    height: 4.5rem;
  }

  .sign {
    width: 2rem;
    height: 2rem;
    margin: 0 2rem;
  }
`;
