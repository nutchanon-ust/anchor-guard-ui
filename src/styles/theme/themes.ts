const lightTheme = {
  primary: 'rgba(215,113,88,1)',
  text: 'rgba(58,52,51,1)',
  textSecondary: 'rgba(58,52,51,0.7)',
  background: 'rgba(255,255,255,1)',
  backgroundVariant: 'rgba(251,249,249,1)',
  border: 'rgba(58,52,51,0.12)',
  borderLight: 'rgba(58,52,51,0.05)',
};

const darkTheme: Theme = {
  primary: 'rgba(220,120,95,1)',
  text: 'rgba(241,233,231,1)',
  textSecondary: 'rgba(241,233,231,0.6)',
  background: 'rgba(0,0,0,1)',
  backgroundVariant: 'rgba(28,26,26,1)',
  border: 'rgba(241,233,231,0.15)',
  borderLight: 'rgba(241,233,231,0.05)',
};

export type Theme = typeof lightTheme;

export const themes = {
  light: lightTheme,
  dark: darkTheme,
};
