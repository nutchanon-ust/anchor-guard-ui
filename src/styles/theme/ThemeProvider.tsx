import * as React from 'react';
import { ThemeProvider as OriginalThemeProvider } from 'styled-components';
import { useSelector } from 'react-redux';
import { useThemeSlice } from './slice';
import { selectTheme } from './slice/selectors';

export const ThemeProvider = (props: { children: React.ReactChild }) => {
  useThemeSlice();

  const theme = useSelector(selectTheme);
  return (
    <OriginalThemeProvider theme={theme}>
      {React.Children.only(props.children)}
    </OriginalThemeProvider>
  );
};
