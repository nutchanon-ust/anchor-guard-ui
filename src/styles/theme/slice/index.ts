import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { useInjectReducer } from 'utils/redux-injectors';
import { getThemeFromStorage } from '../utils';
import { ThemeKeyType, ThemeState } from './types';

export const initialState: ThemeState = {
  selected: getThemeFromStorage() || 'system',
};

const slice = createSlice({
  name: 'theme',
  initialState,
  reducers: {
    changeTheme(state, action: PayloadAction<ThemeKeyType>) {
      state.selected = action.payload;
    },
  },
});

export const { actions: themeActions, reducer } = slice;

export const useThemeSlice = () => {
  useInjectReducer({ key: slice.name, reducer: slice.reducer });
  return { actions: slice.actions };
};
