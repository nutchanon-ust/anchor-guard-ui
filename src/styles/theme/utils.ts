import { ThemeKeyType } from './slice/types';

/* istanbul ignore next line */
export const isSystemDark = window?.matchMedia
  ? window.matchMedia('(prefers-color-scheme: dark)')?.matches
  : undefined;

export function saveTheme(theme: ThemeKeyType) {
  window.localStorage && localStorage.setItem('selectedTheme', theme);
}

/* istanbul ignore next line */
export function getThemeFromStorage(): ThemeKeyType | null {
  return window.localStorage
    ? (localStorage.getItem('selectedTheme') as ThemeKeyType) || null
    : null;
}
