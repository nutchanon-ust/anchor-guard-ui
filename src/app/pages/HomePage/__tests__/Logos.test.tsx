import * as React from 'react';
import { Logos } from '../Logos';
import { render } from '@testing-library/react';
import { ThemeProvider, DefaultTheme } from 'styled-components';

import { themes } from 'styles/theme/themes';

const renderLogos = (theme: DefaultTheme) =>
  render(
    <ThemeProvider theme={theme}>
      <Logos />
    </ThemeProvider>,
  );
describe('<Logos />', () => {
  it('should render and match the snapshot', () => {
    const logos = renderLogos(themes.light);
    expect(logos.asFragment()).toMatchSnapshot();
  });

  it('should use theme from props', () => {
    let logos = renderLogos(themes.light);
    expect(logos.container.firstChild).toHaveStyle(
      `color: ${themes.light.border}`,
    );
    logos = renderLogos(themes.dark);
    expect(logos.container.firstChild).toHaveStyle(
      `color: ${themes.dark.border}`,
    );
  });
});
