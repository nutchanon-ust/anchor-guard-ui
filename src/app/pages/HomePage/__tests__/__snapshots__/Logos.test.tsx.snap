// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logos /> should render and match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(58,52,51,0.12);
}

.c0 .logo {
  width: 4.5rem;
  height: 4.5rem;
}

.c0 .sign {
  width: 2rem;
  height: 2rem;
  margin: 0 2rem;
}

<div
    class="c0"
  >
    <svg
      class="logo"
    >
      cra-logo.svg
    </svg>
    <svg
      class="sign"
    >
      plus-sign.svg
    </svg>
    <svg
      class="logo"
    >
      rp-logo.svg
    </svg>
  </div>
</DocumentFragment>
`;
